src/commands/cmd_bulk_insert.c

```diff
@@ -10,24 +10,53 @@
 #include "../util/thpool/pools.h"
 #include "../bulk_insert/bulk_insert.h"
 
-// bulk-insert context object
-typedef struct {
-	int argc;                      // number of arguments
-	RedisModuleString **argv;      // arguments
-	RedisModuleBlockedClient *bc;  // blocked client
-} BulkCtx;
+#define BATCH_SIZE_LIMIT 1000000 // 1M
+
+// process HEADER token if exists
+static int _Graph_Bulk_Header(RedisModuleCtx *ctx, RedisModuleString ***argv,
+		int *argc, Graph *g) {
+	const char *token = RedisModule_StringPtrLen(**argv, NULL);
+	bool header = strcmp(token, "HEADER") == 0;
+	if(header) {
```

Comment:
"HEADER" is a good way to pre-allocate the graph while maintaining backwards compatibility, I like it.

Since there is no work to be done in this function without it, I'd probably favor the construction:
```
if(!header) return BULK_OK;
```

Reply:
👍